### DISCUSSION TITLE
Eslint config upgrade

### DISCUSSION THREAD

**Discussion by rexkenley** [nrwl/nx#26435] [Questions]
It seems like the new 19.2 is using the new eslint.config.js

Is there a guide on how to upgrade from the old .eslintrc to the new eslint.config.js? Where is eslint.config.js located? I don't see that file after the upgrade.

![image](https://github.com/nrwl/nx/assets/10548251/4ced1c4a-6a4b-4539-a99a-c584190bbaae)

Update: My bad, nx 19.2 is still using eslint 8. So I need some guidance on how to upgrade to eslint 9. But I did see this issue.
https://github.com/nrwl/nx/issues/22576

**Comment by KeithGillette** [comment]
Also seeking guidance on upgrading an Nx (20.3.0) monorepo (with multiple applications and libraries and multiple ESLint plugins) to the new flat configuration file format and from ESLint 8 to 9. The [@nx/eslint:convert-to-flat-config generator ](https://nx.dev/nx-api/eslint/generators/convert-to-flat-config) has not produced a working configuration (and does not modify project-level `.eslintrc.json` files) and my attempts to start from scratch updating the workspace to ESLint 9 have all met with one error or another.

**Reply by scottrepreneur** [reply]
~~Were you able to resolve this?~~ Will give it another go

Ah looks like this was mentioned in https://github.com/nrwl/nx/issues/22576 at [this comment](https://github.com/nrwl/nx/issues/22576#issuecomment-2630924501). Became available in 20.4
https://github.com/nrwl/nx/releases/tag/20.4.0

**Comment by rexkenley** [comment]
https://eslint.org/blog/2024/05/eslint-configuration-migrator/

**Comment by dieguezz** [comment]
In case you want to apply this config to each lib/project inside yout monorepo:
``` 
find . -name ".eslintrc.json" -type f -print0 | while IFS= read -r -d '' file; do
  echo "Migrating $file"
  npx @eslint/migrate-config "$file"
done
```
